File: src/main/java/infinitespire/relics/Freezer.java
package infinitespire.relics;

import com.megacrit.cardcrawl.actions.defect.ChannelAction;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.orbs.Frost;
import com.megacrit.cardcrawl.relics.AbstractRelic;

import infinitespire.abstracts.Relic;

public class Freezer extends Relic {

	public static final String ID = "Freezer";
	
	public Freezer() {
		super(ID, "freezer", RelicTier.UNCOMMON, LandingSound.CLINK);
	}
	
	@Override
	public AbstractRelic makeCopy() {
		return new Freezer();
	}

	@Override
	public void atBattleStart() {
		flash();
		AbstractDungeon.actionManager.addToTop(new ChannelAction(new Frost()));
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/Freezer.kt
package com.blanktheevil.infinitespire.relics

import com.blanktheevil.infinitespire.extensions.makeID
import com.megacrit.cardcrawl.actions.defect.ChannelAction
import com.megacrit.cardcrawl.orbs.Frost

class Freezer : Relic(ID, IMG, TIER, SOUND) {
  companion object {
    val ID = "Freezer".makeID()
    private const val IMG = "freezer"
    private val TIER = RelicTier.COMMON
    private val SOUND = LandingSound.CLINK
  }

  override fun atBattleStart() {
    flash()
    addToBot(ChannelAction(Frost()))
  }
}